lib/kitchen/patches/nokogiri.rb

```diff
@@ -74,6 +74,14 @@ def previous
         prev.text? ? prev.previous : prev
       end
 
+      def preceded_by_text?
+        prev = previous_sibling
+        while !prev.nil? && prev.blank? do prev = prev.previous_sibling end
```

Comment:
this is a good way of getting through a bunch of blank nodes! it doesn't seem likely that the while loop could end up looping infinitely, but it might be a good thing to check with a few more test cases.